time_tracker_api/activities/activities_model.py

```diff
@@ -39,12 +41,72 @@ def __str___(self):
         return "the activity \"%s\"" % self.name  # pragma: no cover
 
 
-def create_dao() -> ActivityDao:
-    repository = CosmosDBRepository.from_definition(container_definition,
-                                                    mapper=ActivityCosmosDBModel)
+class ActivityCosmosDBRepository(CosmosDBRepository):
+    def __init__(self):
+        CosmosDBRepository.__init__(
+            self,
+            container_id=container_definition['id'],
+            partition_key_attribute='tenant_id',
+            mapper=ActivityCosmosDBModel,
+        )
+
+    def create_sql_in_condition(self, id_list):
+        id_values = self.convert_list_to_tuple_string(id_list)
+
+        return "c.id IN {value_condition}".format(value_condition=id_values)
+
+    def convert_list_to_tuple_string(self, id_list):
+        self.validate_list(id_list)
+        id_value = (
+            f"('{id_list[0]}')" if len(id_list) == 1 else str(tuple(id_list))
+        )
+        return id_value
+
+    def validate_list(self, id_list):
+        assert isinstance(id_list, list)
+        assert len(id_list) > 0
```

Comment:
These methods seem generic, what if you move them to the `utils` folder to be able to use them in other classes, do you think it is possible or do they use a specific logic for this class?

Reply:
It's a nice idea, @eduardisrael will implement something similar, so he will need that functions.

Reply:
https://github.com/ioet/time-tracker-backend/blob/c94b511ba5720753b273fd87fb6bb2bafb4e8851/utils/repository.py#L1-L11

@israteneda  now it is implemented!